### DISCUSSION TITLE
How can I highlight specific lines in decompiler window through GhidraScript?

### DISCUSSION THREAD

**Discussion by aa-software2112** [NationalSecurityAgency/ghidra#5294] [General]
Hello!

I would like to know if there is supported API to highlight specific lines in the decompiler window through a GhidraScript (say, by providing the address of the equivalent line to highlught, or even by providing a varnode used on that line).

Appreciate the help.

Anthony

**Comment by dragonmacher** [comment]
There is a service to do this, the `DecompilerHighlightService`.  See this ticket for a discussion and an example: https://github.com/NationalSecurityAgency/ghidra/issues/3749#issuecomment-992973502

The comment just above this comment shows the code for `MyMatcher`.
